### ISSUE TITLE
minimizeAutoencoder

### ISSUE THREAD

**Issue by j0ntt** [maddin79/darch#4]
There is apparently a bug in minimizeAutoencoder and probably minimizeClassifier as well. 

This error keeps showing.
 
Error in data * log(output) : non-conformable arrays

backpropagation and rpropagation work just fine.

Perhaps I could try using these straight

http://www.cs.toronto.edu/~hinton/MatlabForSciencePaper.html

Those are the original codes.

But saviola777 told me in darch2 issue discussion channel that he might try to fix it.

**Comment by saviola777** [comment]
Just for clarification: are you using the CRAN darch version or a git version (and if so, which one)?

I did a quick test using minimizeClassifier with the current git version. After adjusting it to the new interface, it worked without problems. Will test minimizeAutoencoder later on, and will test using the CRAN version as well, although the latter is less relevant to me, as I plan to release version 1.0 to CRAN within the next month or so, so I'll focus on fixing it going from the current git version.

**Comment by j0ntt** [comment]
Actually the current CRAN version. I will try the git version next.

**Comment by saviola777** [comment]
Alright, I'll have to adjust some things in both functions before they can work in the current git version. I hope you like the new interface when you try the git version. I'll report back once I tested and fixed both functions in the current git version.

**Comment by saviola777** [comment]
The error

    Error in data * log(output) : non-conformable arrays

is probably the result of an invalid network structure (at least for the autoencoder function), as an autoencoder (as far as I understand) is supposed to reconstruct the input after compressing it in some smaller "code layer", meaning that the first (input) and last (output) layer have to have the same number of neurons.

But even then, while the algorithm works (no more errors), it doesn't seem to produce useful results (the MSE stays constant). Unfortunately, my knowledge of these algorithms is rather limited.

The minimizeClassifier fine-tuning works now, see the [example](https://github.com/maddin79/darch/blob/master/inst/examples/example.cg.R) for details.

**Comment by j0ntt** [comment]
I tried minimizeAutoencoder with structure 100,50,10,50,100 (the data has 100 variables). It gives a different error this time. length = 5

INFO [2015-11-07 20:16:35] Start deep architecture fine-tuning
INFO [2015-11-07 20:16:35] Number of Batches:  1
[1] "Linesearch 1 Value 63.421801870939"
[1] "Linesearch 2 Value 63.3840077259058"
[1] "Linesearch 3 Value 63.3495287047134"
[1] "Linesearch 4 Value 63.348814529677"
[1] "Linesearch 5 Value 63.3484015249208"
Error in original[] - estimate[] : non-conformable arrays

The same happened with 100,50,1,50,100.

The targetData parameter is 1-dimensional.  I tried this with the CRAN version.
